lib/tasks/bower-install.js

```diff
@@ -1,32 +1,48 @@
 'use strict';
 
 // Runs `bower install` in cwd
+const fs = require('fs-extra');
 const path = require('path');
 const existsSync = require('exists-sync');
 const execa = require('execa');
 const RSVP = require('rsvp');
 const Task = require('../models/task');
 const formatPackageList = require('../utilities/format-package-list');
 
+const logger = require('heimdalljs-logger')('ember-cli:tasks:bower-install');
+
 const Promise = RSVP.Promise;
 const resolve = RSVP.denodeify(require('resolve'));
+const writeJson = RSVP.denodeify(fs.writeJson);
 
 const cliPath = path.resolve(`${__dirname}/../..`);
 
 class BowerInstallTask extends Task {
 
+  resolveBower() {
+    logger.info('Resolving "bower" from %s ...', cliPath);
+    return resolve('bower', { basedir: cliPath });
+  }
+
+  importBower(path) {
+    logger.info('Importing "bower" from: %s', path);
+    this.bower = require(path);
+  }
+
   ensureBower() {
     if (this.bower) {
-      return Promise.resolve(this.bower);
+      return Promise.resolve();
     }
 
-    return resolve('bower', { basedir: cliPath })
+    return this.resolveBower()
       .catch(() => this.installBower()
-        .then(() => resolve('bower', { basedir: cliPath })))
-      .then(bowerPath => require(bowerPath));
+        .then(() => this.resolveBower()))
+      .then(bowerPath => this.importBower(bowerPath));
   }
 
   installBower() {
+    logger.info('Installing "bower" via NPM into: %s', cliPath);
```

Comment:
There are several issues with what you're proposing since we're not shelling out to bower but requiring it instead. Also using global bower seems unrelated to this PR and can be discussed separately.